### PR TITLE
CI: retry setup on failure

### PR DESCRIPTION
**Pull request by jnewb1**
Tried using wretry.action, but it doesn't support retrying local actions due to some github actions constraints

Checkout cannot be done in a setup script because it requires the .github folder to be checked out first, so we might need a different solution for that

### CODE REVIEW

**Comment by jnewb1** [comment]
example where I forced a failure 3 times, whole job fails:
https://github.com/commaai/openpilot/actions/runs/6091027666/job/16526955774?pr=29785

example where only 1 job fails: retried and succeeded:
https://github.com/commaai/openpilot/actions/runs/6090987934/job/16526843942

**Comment by jnewb1** [comment]
@adeebshihadeh this is a bit hacky but it seems to be the only solution for retrying local composite actions

File: .github/workflows/base_setup/action.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: 'openpilot env setup'
+
```
Comment (adeebshihadeh) [inline .github/workflows/base_setup/action.yaml:2]: let's keep this one called "setup", it's not really a base setup

File: .github/workflows/setup/action.yaml [hunk 2]
```diff
@@ -1,51 +1,36 @@
-name: 'openpilot env setup'
+name: 'openpilot env setup, with retry on failure'
```
Comment (adeebshihadeh) [inline .github/workflows/setup/action.yaml:2]: let's call this setup-with-retry

File: .github/workflows/base_setup/action.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+inputs:
```
Comment (adeebshihadeh) [inline .github/workflows/base_setup/action.yaml:3]: can you add an extra input that we assert is set? then it's hard to accidentally use this setup action without the retry

File: .github/workflows/setup/action.yaml (hunk 2 above)
Comment (adeebshihadeh) [reply .github/workflows/setup/action.yaml:2]: or can we do a nested action like [actions/cache](https://github.com/actions/cache)

File: .github/workflows/base_setup/action.yaml (hunk 3 above)
Comment (jnewb1) [reply .github/workflows/base_setup/action.yaml]: Done

File: .github/workflows/setup/action.yaml (hunk 2 above)
Comment (jnewb1) [reply .github/workflows/setup/action.yaml]: Renamed to setup-with-retry

File: .github/workflows/base_setup/action.yaml (hunk 1 above)
Comment (jnewb1) [reply .github/workflows/base_setup/action.yaml]: Done

**Review by adeebshihadeh — APPROVED** [review]
LGTM. separately, can we do something about the PR comments job so it doesn't fail in these cases?

**Pull request merged**